File: benchmarks/oopsla25/subjects/014.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

char* f014(const char *s) {
    if (strlen(s) != 1) {
        fprintf(stderr, "Exception\n");
        exit(1);
    }
    char *result = malloc(2);
    strncpy(result, s, 1);
    result[1] = '\0';
    return result;
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  char *s = f014(my_string);
  printf("%s\n", s);
  return 0;
}

File: tests/oopsla25/014.py
#!/usr/bin/env python3

import sys

def f014(s: str):
  if not len(s) == 1:
    raise Exception
  return s[0:1]

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  print(f014(my_string))
